mailbox/backup/src/test/java/org/apache/james/mailbox/backup/InternalDateExtraFieldTest.java

```diff
@@ -0,0 +1,330 @@
+            byte[] input = new byte[9];
+
+            assertThatThrownBy(() -> testee.parseFromCentralDirectoryData(input, 0, 9))
+                .isInstanceOf(ZipException.class);
+        }
+
+        @Test
+        void parseFromCentralDirectoryDataShouldParseWhenZero() throws Exception {
+            InternalDateExtraField testee = new InternalDateExtraField();
+
+            testee.parseFromCentralDirectoryData(ZERO_AS_BYTE_ARRAY, 0, 8);
+            assertThat(testee.getValue())
+                .contains(0L);
+        }
+
+        @Test
+        void parseFromCentralDirectoryDataShouldParseWhen123456789ABCDEF0InLittleEndian() throws Exception {
+            InternalDateExtraField testee = new InternalDateExtraField();
+
+            testee.parseFromCentralDirectoryData(_123456789ABCDEF0_AS_LE_BYTE_ARRAY, 0, 8);
+            assertThat(testee.getValue())
+                .contains(0x123456789ABCDEF0L);
+        }
+
+        @Test
+        void parseFromCentralDirectoryDataShouldParseWhenFEDCBA9876543210InLittleEndian() throws Exception {
+            InternalDateExtraField testee = new InternalDateExtraField();
+            byte[] input = FEDCBA9876543210_AS_LE_BYTE_ARRAY;
+
+            testee.parseFromCentralDirectoryData(input, 0, 8);
+            assertThat(testee.getValue())
+                .contains(0xFEDCBA9876543210L);
+        }
+
+        @Test
+        void parseFromCentralDirectoryDataShouldHandleOffset() throws Exception {
+            InternalDateExtraField testee = new InternalDateExtraField();
+            byte[] input = Arrays.concatenate(UNUSED, _123456789ABCDEF0_AS_LE_BYTE_ARRAY);
+
+            testee.parseFromCentralDirectoryData(input, 2, 8);
+            assertThat(testee.getValue())
+                .contains(0x123456789ABCDEF0L);
+        }
+
+        @Test
+        void parseFromCentralDirectoryDataShouldReturnZeroDayWhenZero() throws Exception {
+            InternalDateExtraField testee = new InternalDateExtraField();
+
+            testee.parseFromCentralDirectoryData(ZERO_AS_BYTE_ARRAY, 0, 8);
+
+            assertThat(testee.getDateValue())
+                .contains(new Date(0L));
+        }
+
+        @Test
+        void parseFromCentralDirectoryDataShouldReturnDefaultDateWhenPassDefaultUTCDateByteArray() throws Exception {
+            InternalDateExtraField testee = new InternalDateExtraField();
+            testee.parseFromCentralDirectoryData(DEFAULT_DATE_BYTE_ARRAY, 0, 8);
+
+            assertThat(testee.getDateValue())
+                .contains(DEFAULT_DATE_UTC);
+        }
+
+        @Test
+        void parseFromCentralDirectoryDataShouldReturnLocalDateWhenPassDefaultDateByteArray() throws Exception {
+            InternalDateExtraField testee = new InternalDateExtraField();
+            testee.parseFromCentralDirectoryData(DEFAULT_DATE_BYTE_ARRAY, 0, 8);
+
+            assertThat(testee.getLocalDateValue())
+                .contains(DEFAULT_DATE_LOCAL);
+        }
+    }
+}
```

Comment:
I would have expected some specific date test cases in this class

Reply:
I can't get it. You want to have more different days - edge cases as input of our tests?

Reply:
In these tests you are just checking that a long is correctly converted to a byte array.
I would like some tests to show that a date is well converted, and that when you get a byte array you get the same date.

Reply:
We miss a test showing that "default date" produces default byte array